helm/resource_release_test.go

```diff
@@ -433,6 +453,78 @@ func TestAccResourceRelease_updateVersionFromRelease(t *testing.T) {
 	})
 }
 
+func TestAccResourceRelease_namespaceDoesNotExist(t *testing.T) {
+	name := fmt.Sprintf("test-namespace-does-not-exist-%s", acctest.RandString(10))
+	namespace := fmt.Sprintf("%s-%s", testNamespace, acctest.RandString(10))
+
+	defer deleteNamespace(t, namespace)
+
+	broken := fmt.Sprintf(`
+	resource "helm_release" "test" {
+		name        = %q
+		namespace   = "does-not-exist"
+		repository  = "https://kubernetes-charts.storage.googleapis.com"
+		chart       = "nginx-ingress"
+		set {
+			name = "controller.service.type"
+			value = "ClusterIP"
+		}
+	}`, name)
+
+	fixed := fmt.Sprintf(`
+	resource "helm_release" "test" {
+		name        = %q
+		namespace   = %q
+		repository  = "https://kubernetes-charts.storage.googleapis.com"
+		chart       = "nginx-ingress"
+		set {
+			name = "controller.service.type"
+			value = "ClusterIP"
+		}
+	}`, name, namespace)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t, namespace) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckHelmReleaseDestroy(namespace),
+		Steps: []resource.TestStep{{
+			Config:             broken,
+			ExpectError:        regexp.MustCompile(`failed to create: namespaces "does-not-exist" not found`),
+			ExpectNonEmptyPlan: true,
+		}, {
+			Config: fixed,
+			Check: resource.ComposeAggregateTestCheckFunc(
+				resource.TestCheckResourceAttr("helm_release.test", "status", release.StatusDeployed.String()),
+			),
+		}},
+	})
+}
+
+func TestAccResourceRelease_invalidName(t *testing.T) {
+	namespace := fmt.Sprintf("%s-%s", testNamespace, acctest.RandString(10))
+
+	defer deleteNamespace(t, namespace)
+
+	broken := fmt.Sprintf(`
+	resource "helm_release" "test" {
+		name        = "this_should_not_work"
+		namespace   = %q
+		repository  = "https://kubernetes-charts.storage.googleapis.com"
+		chart       = "nginx-ingress"
+	}`, namespace)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t, namespace) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckHelmReleaseDestroy(namespace),
+		Steps: []resource.TestStep{{
+			Config:             broken,
+			ExpectError:        regexp.MustCompile("create: failed to create"),
```

Comment:
There is a bug in helm at the moment (https://github.com/helm/helm/issues/3493) where it does not validate the release name before trying to create the ConfigMap for it, so you get an error from deep in the bowels of k8s here. I've left this test case vague because I expect the error message will change to something more human friendly soon.